juju_verify/verifiers/base.py

```diff
@@ -198,26 +197,14 @@ def check_has_sub_machines(self) -> Result:
                         child=potential_child, parent=parent_unit
                     )
 
-        task_map = {
-            child_tag: parent_child_pair.child.is_leader_from_status()
-            for child_tag, parent_child_pair in parent_child_pairs.items()
-        }
-
-        loop = asyncio.get_event_loop()
-
-        results = loop.run_until_complete(asyncio.gather(*task_map.values()))
-        result_map = dict(zip(task_map.keys(), results))
-
```

Comment:
This change seems unrelated to the removal of the event loop. What's the story here?

Reply:
Ah sorry, you are right. This change should not be here.

This is used to get knowledge if unit is leader and that use it to provide `*` in unit name.
e.g.  ```bash
# with information about leader
[WARN] ceph-osd/0 has units running on child machines: ceph-mon/0*
# without the information
[WARN] ceph-osd/0 has units running on child machines: ceph-mon/0
```
IMO:  This is just waste of time to collect this information. It's not really useful.

Reply:
I also don't quite understand the need to check for unit leader. Maybe @xavpaice could provide some insight on this as he designed this check?

Reply:
I have reversed these changes and we can redo it in another PR.